stdlib/public/core/FloatingPointParsing.swift.gyb

```diff
@@ -53,63 +53,91 @@ extension ${Self}: LosslessStringConvertible {
+  /// - A *decimal string* contains a significand consisting of one
+  ///   or more decimal digits that may include a decimal point:
   ///
   ///       let c = ${Self}("-1.0")
   ///       // c == -1.0
   ///
   ///       let d = ${Self}("28.375")
   ///       // d == 28.375
   ///
-  ///   A decimal value may also include an exponent following the significand,
-  ///   indicating the power of 10 by which the significand should be
-  ///   multiplied. If included, the exponent is separated by a single
+  ///   A decimal string may also include an exponent following the
+  ///   significand, indicating the power of 10 by which the significand should
+  ///   be multiplied. If included, the exponent is separated by a single
   ///   character, `e` or `E`, and consists of an optional plus or minus sign
   ///   character and a sequence of decimal digits.
   ///
   ///       let e = ${Self}("2837.5e-2")
   ///       // e == 28.375
   ///
-  /// - A *hexadecimal value* contains the significand, either `0X` or `0x`,
-  ///   followed by a sequence of hexadecimal digits. The significand may
+  ///   A decimal string is converted to a ${Self}
+  ///   instance using the IEEE 754 roundTiesToEven (default) rounding
+  ///   attribute.
+  ///   Very small values are rounded to positive or
+  ///   negative zero, very large values are rounded
+  ///   to plus or minus infinity.
+  ///
+  ///       let y = ${Self}("1.23e-9999")
+  ///       // y == 0.0
+  ///       // y?.sign == .plus
+  ///
+  ///       let z = ${Self}("-7.89e-7206")
+  ///       // z == -0.0
+  ///       // z?.sign == .minus
+  ///
+  ///       let r = ${Self}("1.23e17802")
+  ///       // r == ${Self}.infinity
+  ///
+  ///       let s = ${Self}("-7.89e7206")
+  ///       // s == ${Self}.-infinity
+  ///
+  /// - A *hexadecimal string* contains a significand consisting of
+  ///   `0X` or `0x` followed by one or more hexadecimal digits that may
   ///   include a decimal point.
   ///
   ///       let f = ${Self}("0x1c.6")
   ///       // f == 28.375
   ///
-  ///   A hexadecimal value may also include an exponent following the
-  ///   significand, indicating the power of 2 by which the significand should
+  ///   A hexadecimal string may also include an exponent
+  ///   indicating the power of 2 by which the significand should
   ///   be multiplied. If included, the exponent is separated by a single
   ///   character, `p` or `P`, and consists of an optional plus or minus sign
   ///   character and a sequence of decimal digits.
   ///
   ///       let g = ${Self}("0x1.c6p4")
   ///       // g == 28.375
   ///
-  /// - A value of *infinity* contains one of the strings `"inf"` or
-  ///   `"infinity"`, case insensitive.
+  ///   As with decimal strings, hexadecimal strings are rounded
+  ///   to the closest ${Self} instance to the real number.  Very small
+  ///   or very large values are rounded to plus or minus zero or plus
+  ///   or minus infinity.
```

Comment:
```suggestion
  ///   or minus infinity, respectively.
```

Perhaps "very small" and "very large" can be clarified in one place instead of twice in this documentation. Also, perhaps using the ± sign here would improve clarity of the text.

Reply:
Rather than duplicating the rounding/underflow/overflow discussion for both decimal and hexadecimal, I instead consolidated that at the end after the bullet list of formats.